File: dohyeon/Programmers/Level_0/Lv0_특이한_정렬.py

def solution(numlist, n):
    answer = []
    for i in numlist:
        answer.append(i - n)
    
    result = []
    for i in sorted(answer[:], key=lambda x:[abs(x), -x]): 
        result.append(numlist[answer.index(i)])

    return result

File: seongwon/Programmers/Lv0_특이한_정렬.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;

bool cmp(int a, int b) {     //절대값일때 조건 추가
    if (abs(a) == abs(b)) {
        return a > b;
    }
    return abs(a) < abs(b); 
}

vector<int> solution(vector<int> numlist, int n) {
    vector<int> answer;
    for(int i : numlist){
        answer.push_back(i-n);  //n과의 거리차이 넣음
    }
    sort(answer.begin(), answer.end(), cmp);
    for(int& i : answer){
        i +=n;
    }
    return answer;
}
